docs/src/components/landing/PricingTable.js

```diff
@@ -0,0 +1,451 @@
+          theme.palette.mode === 'dark'
+            ? alpha(theme.palette.primaryDark[900], 0.72)
+            : 'rgba(255,255,255,0.72)',
+      }}
+    >
+      <Container
+        sx={{
+          display: 'grid',
+          gridTemplateColumns: `minmax(160px, 1fr) repeat(${plans.length}, minmax(240px, 1fr))`,
+        }}
+      >
+        <Typography variant="body2" fontWeight="bold" sx={{ px: 2, py: 1 }}>
+          Plans
+        </Typography>
+        {plans.map((plan) => (
+          <Box key={plan} sx={{ px: 2, py: 1 }}>
+            <PlanName planInfo={planInfo[plan]} centered disableDescription />
+          </Box>
+        ))}
+      </Container>
+    </Box>
+  );
+};
+
+StickyHead.propTypes = {
+  container: PropTypes.shape({
+    current: PropTypes.shape({
+      getBoundingClientRect: PropTypes.func.isRequired,
+    }),
+  }).isRequired,
+  disableCalculation: PropTypes.bool,
+  planInfo: PropTypes.object,
+  plans: PropTypes.arrayOf(PropTypes.string).isRequired,
+};
+
+const divider = <Divider />;
+
+function PricingTable({
+  columnHeaderHidden,
+  planInfo,
+  plans,
+  rowHeaders,
+  communityData,
+  commercialData,
+  ...props
+}) {
+  const tableRef = React.useRef(null);
+  const gridSx = {
+    display: 'grid',
+    gridTemplateColumns: `minmax(160px, 1fr) repeat(${plans.length}, minmax(${
+      columnHeaderHidden ? '0px' : '240px'
+    }, 1fr))`,
+  };
+  function renderRow(key) {
+    return (
+      <Box
+        sx={{
+          ...gridSx,
+          '&:hover': {
+            bgcolor: (theme) =>
+              theme.palette.mode === 'dark'
+                ? alpha(theme.palette.primaryDark[900], 0.3)
+                : alpha(theme.palette.grey[50], 0.4),
+            '@media (hover: none)': {
+              bgcolor: 'initial',
+            },
+          },
+        }}
+      >
+        {rowHeaders[key]}
+        {plans.map((id, index) => (
+          <Cell key={id} highlighted={index % 2 === 1}>
+            {id === 'community' ? communityData[key] : null}
+            {id === 'commercial' ? commercialData[key] : null}
+          </Cell>
+        ))}
+      </Box>
+    );
+  }
+  return (
+    <Box
+      ref={tableRef}
+      {...props}
+      sx={{
+        width: '100%',
+        overflow: 'auto',
+        py: { xs: 2, md: 4 },
+        ...props.sx,
+      }}
+    >
+      <StickyHead
+        plans={plans}
+        planInfo={planInfo}
+        container={tableRef}
+        disableCalculation={columnHeaderHidden}
+      />
+      {!columnHeaderHidden ? (
+        <Box sx={gridSx}>
+          <Typography variant="body2" fontWeight="bold" sx={{ p: 2 }}>
+            Plans
+          </Typography>
+          {plans.map((plan) => (
+            <Box key={plan} sx={{ display: 'flex', flexDirection: 'column', p: 2 }}>
+              <PlanName planInfo={planInfo[plan]} />
+            </Box>
+          ))}
+        </Box>
+      ) : null}
+      <RowHead>Apps</RowHead>
+      {renderRow('Self-hosting')}
```

Comment:
we could probably map this from some array to have less repetition

Reply:
After looking at it more, I feel it's fine to unbundle the rendering and the content this way since it allows more flexibility